iFixit/react-commerce
# feat: project setup

Brace yourselves, this is a long PR. I'll try to summarize here to make it easier to parse the code.

## Goal
The main goal of this PR is to setup a project to enable code review of the prototype we were writing in CodeSandbox.

## Project setup

The project is setup using Next.js, so down the road we can test server side rendering, and start to see how a headless implementation looks like with Shopify and Algolia.

## Features

- **Implementation of the collection page.** Many details are still missing, but the major pieces are there
- **"Parts" collection page.** Enough is implemented to enable navigation, while search and "Popular fix kits" have been left out of this PR
- **Navigation between collections.** It is possible to navigate collections using subcategories and breadcrumbs. The data is fetched from Shopify Test store. 
  > ⚠️ Only "leaf" collections contain products right now, as intermediate collections are empty in Shopify
- **Algolia hooks.** Algolia hooks are currently implemented under the `/libs` folder, so it will be easy to extract them into a separate package when their API has stabilized (support for SSR and using multiple indexes for sorting is not implemented yet).

> Note: the navigation bar serves only the purpose of giving structure to the page

## QA

1. Clone this repo 
2. Install dependencies with `npm install`
3. Run the dev server with `npm run dev`
4. Ensure Next.js app starts successfully
5. Ensure it is possible to navigate between pages using subcategories and breadcrumbs.
6. Navigate to http://localhost:3000/collections/tools and check that filters and search are working

> ⚠️ The implementation is not supposed to be pixel perfect right now, many things are still missing. For example there's no optimization around images.
